zebra-chain/src/transaction.rs

```diff
@@ -208,7 +209,13 @@ impl Transaction {
                 shielded_data: Some(shielded_data),
                 ..
             } => Box::new(shielded_data.nullifiers()),
+            Transaction::V5 { .. } => {
+                unimplemented!("v5 transaction format as specified in ZIP-225")
+            }
             // No JoinSplits
+            Transaction::V1 { .. } | Transaction::V2 { .. } | Transaction::V3 { .. } => {
+                Box::new(std::iter::empty())
+            }
             _ => Box::new(std::iter::empty()),
```

Comment:
Let's match the full pattern here exhaustively:
```suggestion
            Transaction::V1 { .. } | Transaction::V2 { .. } | Transaction::V3 { .. } | Transaction::V4 {
                shielded_data: None,
                ..
            } => {
                Box::new(std::iter::empty())
            }
```